### DISCUSSION TITLE
Add GetSinglePixelSpanMethod

### DISCUSSION THREAD

**Discussion by UkooLabs** [SixLabors/ImageSharp#1247] [Ideas]
Please add a method something like as finding myself adding to my code and think would be useful in ImageSharp..

```
        public static Span<TPixel> GetSinglePixelSpan<TPixel> (this Image<TPixel> image) where TPixel : unmanaged, IPixel<TPixel>
        {
            if (!image.TryGetSinglePixelSpan(out var pixelSpan))
            {
                throw new InvalidMemoryOperationException("Unable to get pixelspan.");
            }
            return pixelSpan;
        }
```

**Comment by JimBobSquarePants** [comment]
Interesting... we deliberately made it a Boolean so people weren’t surprised when it failed. What’s the advantage of throwing for you?

**Reply by UkooLabs** [reply]
@JimBobSquarePants Example could be something like vendrid imagesharp, when trying to convert pixel data to a GPU texture I could return null if failed but that doesn’t give any kind of feedback to user using library as to why could be null, where as a exception being thrown can be handled. If was my own method I would use the try but since would be used in a method used by existing users and don’t want to make changes to method structures  the non try way would be useful and save me throwing my own exception

**Reply by JimBobSquarePants** [reply]
I see, thanks @antonfirsov what do you think?

**Reply by antonfirsov** [reply]
I think majority of users would miss the fact that `GetSinglePixelSpan` can fail. This will lead to rare sporadic unhandled exceptions which are hard to understand and debug for them. I'd rather keep this particular API fool-proof.

@UkooLabs is there anything preventing to add the given helper method on library consumer side?

**Reply by UkooLabs** [reply]
@antonfirsov no nothing stopping me at all I’m just used to where a try exists there’s a non try version e.g. int.parse or int.tryparse for now I’m just using the extension code pasted above so no worries if does not get added :)

**Reply by JimBobSquarePants** [reply]
I've always thought of those as a legacy issue tbh. In my headcanon the non try variants existed first with the try added after.
